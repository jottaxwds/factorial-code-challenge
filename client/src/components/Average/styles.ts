import { CircularProgressbar } from "react-circular-progressbar";
import "react-circular-progressbar/dist/styles.css";
import styled from "styled-components";
import theme from "../../theme";

export const AverageWrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding: ${theme.spacing.small};
  padding-bottom: ${theme.spacing.smallMedium};
  border-radius: 8px;
  background: ${theme.palette.midBlue};
`;

export const Title = styled.h4`
  ${theme.typography.cta};
  color: ${theme.palette.subdued};
  margin: 0 0 ${theme.spacing.medium};
`;

export const Progress = styled(CircularProgressbar)<{ value: number }>`
  &&& {
    .CircularProgressbar {
      width: 30%;
    }
    .CircularProgressbar-path {
      stroke: ${({ value }) => {
        if (value < 30) {
          return theme.palette.green;
        }
        if (value > 30 && value < 70) {
          return theme.palette.orange;
        }
        return theme.palette.red;
      }};
    }
  }
`;
