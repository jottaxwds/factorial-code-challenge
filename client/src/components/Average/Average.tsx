import * as S from "./styles";

export interface AverageProps {
  title: string;
  value: number;
  displayValue: string;
}

const Average = ({ title, value, displayValue }: AverageProps) => {
  return (
    <S.AverageWrapper data-testid={"average"}>
      <S.Title>{title}</S.Title>
      <S.Progress
        data-testid={"average-progress"}
        value={value}
        text={displayValue}
      />
    </S.AverageWrapper>
  );
};

export default Average;
