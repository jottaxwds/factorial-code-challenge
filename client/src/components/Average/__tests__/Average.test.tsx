import { render, screen } from "@testing-library/react";
import Average from "../Average";

describe("Average", () => {
  it("Should a Title and CircularProgress by given average data", () => {
    render(<Average title="Average Title" value={90} displayValue={"900"} />);
    expect(screen.getByTestId("average")).toMatchSnapshot();
  });
});
