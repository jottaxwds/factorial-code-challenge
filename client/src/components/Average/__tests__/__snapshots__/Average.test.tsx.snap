// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Average Should a Title and CircularProgress by given average data 1`] = `
<div
  class="sc-beqWaB efcNtM"
  data-testid="average"
>
  <h4
    class="sc-gueYoa hCdPey"
  >
    Average Title
  </h4>
  <svg
    class="CircularProgressbar sc-dmqHEX crukvD"
    data-test-id="CircularProgressbar"
    viewBox="0 0 100 100"
  >
    <path
      class="CircularProgressbar-trail"
      d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
      fill-opacity="0"
      stroke-width="8"
      style="stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 0px;"
    />
    <path
      class="CircularProgressbar-path"
      d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
      fill-opacity="0"
      stroke-width="8"
      style="stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 28.90265241302609px;"
    />
    <text
      class="CircularProgressbar-text"
      x="50"
      y="50"
    >
      900
    </text>
  </svg>
</div>
`;
